pkg/icingadb/schema.go

```diff
@@ -15,27 +17,74 @@ const (
 	expectedPostgresSchemaVersion = 4
 )
 
-// CheckSchema asserts the database schema of the expected version being present.
-func CheckSchema(ctx context.Context, db *database.DB) error {
-	var expectedDbSchemaVersion uint16
+var (
+	// ErrSchemaNotExists implies that no Icinga DB schema has been imported.
+	ErrSchemaNotExists = fmt.Errorf("no database schema exists")
+
+	// ErrSchemaMismatch implies an unexpected schema version, most likely after Icinga DB was updated but the database
+	// missed the schema upgrade.
+	ErrSchemaMismatch = fmt.Errorf("unexpected database schema version")
+)
+
+// CheckSchema verifies the correct database schema is present.
+//
+// This function returns the following error types, possibly wrapped:
+//   - If no schema exists, the error returned is ErrSchemaNotExists.
+//   - If the schema version does not match the expected version, the error returned is ErrSchemaMismatch.
+//   - Otherwise, the original error is returned, e.g., for general database problems.
+func CheckSchema(ctx context.Context, db *database.DB, databaseName string) error {
+	var (
+		tableSchema             string
+		expectedDbSchemaVersion uint16
+	)
 	switch db.DriverName() {
 	case database.MySQL:
+		tableSchema = databaseName
 		expectedDbSchemaVersion = expectedMysqlSchemaVersion
+
 	case database.PostgreSQL:
+		tableSchema = "public"
 		expectedDbSchemaVersion = expectedPostgresSchemaVersion
-	}
 
-	var version uint16
+	default:
+		return fmt.Errorf("unsupported database driver %q", db.DriverName())
+	}
 
+	var tableSchemaCount int
 	err := retry.WithBackoff(
 		ctx,
-		func(ctx context.Context) (err error) {
+		func(ctx context.Context) error {
+			// The following query should return either 0 or 1 depending on the existence of the icingadb_schema table.
```

Comment:
The `SELECT 1` thing was misleading on my part. I somehow thought that `Scan()` was not required, but of course it is, so you can remove the comment.